Makefile

```diff
@@ -1,11 +1,15 @@
 WERROR?=-Werror
 PKGS=sdl2
-# TODO(#118): try to enable -Weverything and disable things that don't make sense
 CFLAGS=-Wall -Wextra $(WERROR) -pedantic -I.
-CXXFLAGS=$(CFLAGS) -std=c++17 -fno-exceptions -Wno-missing-braces -Wswitch-enum `pkg-config --cflags $(PKGS)`
+ifeq ($(OS),Windows_NT)
```

Comment:
Can we check that we are compile with a MinGW compiler instead of compiling on Windows? So far I don't really see any Windows specific changes, but rather MinGW specific ones.

Reply:
I tried all predefined macros, but none of them are working. Maybe this is explained by this [MSYS2 page](https://www.msys2.org/wiki/Porting/):

> The latter one is called `mingw32-make` on command line, is fully native and doesn't depend on msys2 shells. The downside is that it doesn't work with many Makefiles. Unless you know what you're doing, use the regular `make`.

But regular `make` from msys2 shell requires some extra moves. I like that you can just install MSYS2, use it's pacman to install SDL, add msys/mingw64/bin to PATH and use `mingw32-make` from simple cmd.exe. So I might introduced hackish way of building in README.md:
``` console
$ set __MINGW32__=1 && mingw32-make -B
```